### PR TITLE
solution was added

### PR DESCRIPTION
**Pull request by Maksym845**

### CODE REVIEW

**Review by Yeliseiev — APPROVED** [review]
Good job! I think your solution is more convenient than mine🙂

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,78 @@
...
+      ));
+      break;
+
+    case SortType.Married:
+      sortedListOfStudents.sort((person: Student, prevPerson: Student) => (
+        order === 'asc'
+          ? Number(person[sortBy]) - Number(prevPerson[sortBy])
+          : Number(prevPerson[sortBy]) - Number(person[sortBy])
+      ));
+      break;
```
Comment (OleksiiBlezniuk) [inline src/sortStudents.ts:51]: It's a same logic, let's combine cases

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,78 @@
...
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedListOfStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      sortedListOfStudents.sort((person: Student, prevPerson: Student) => (
```
Comment (OleksiiBlezniuk) [inline src/sortStudents.ts:30]: We can move switch/case inside `sort` method and just return needed comparing result, why duplicate this line of code every time? 🙂

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,51 @@
...
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function getAverageValue(numbers: number[]): number {
+  return numbers.reduce((prev, curr) => prev + curr) / numbers.length;
```
Comment (sergii-nosachenko) [inline src/sortStudents.ts:21]:
```suggestion
  return numbers.reduce((prev, curr) => prev + curr, 0) / numbers.length;
```
Add starting value every time, it's safer and bug proof.

**Review by sergii-nosachenko — APPROVED** [review]
Good job 👍
Read the comment.

**Pull request closed without merging**